docs/tex/mcmc.tex

```diff
@@ -0,0 +1,208 @@
+\documentclass{article}
+\usepackage{geometry}
+\usepackage[utf8]{inputenc}
+\usepackage{amssymb}
+\usepackage{amsmath}
+\usepackage{amsthm}
+\usepackage{url}
+
+% bibliography
+\PassOptionsToPackage{numbers, compress}{natbib}
+\usepackage{natbib}
+
+\newcommand{\reals}{\mathbb{R}}
+\newcommand{\normal}{\mathcal{N}}
+\newcommand{\pathx}{x}
+\newcommand{\pathy}{y}
+\newcommand{\measx}{\tilde{x}}
+\newcommand{\measy}{\tilde{y}}
+
+\newtheorem{prop}{Proposition}
+
+\title{Efficient Automated Calculation of the Jacobian Determinant for Reversible Jump MCMC}
+\author{Marco Cusumano-Towner}
+\date{}
+
+\begin{document}
+
+\maketitle
+
+\section{Introduction}
+Reversible jump MCMC~\citep{green1995reversible} (RJMCMC) is a general framework for constructing Markov chain Monte Carlo kernels that satisfy detailed balance, while permitting transdimensional and/or deterministic moves.
+Some generic transdimensional MCMC kernels that are implemented by various probabilistic programming systems, including `single-site resimulation Metropolis-Hastings' ~\citep{goodman2008church,wingate2011lightweight} can be justified as instances of RJMCMC.
+However, it is possible to construct much more efficient MCMC kernels that are specialized to a model using the RJMCMC framework.
+The Gen~\citep{cusumano2019gen} probabilistic programming system allows users to program their own custom RJMCMC kernels that are specialized to the model, by specifying an auxiliary generative function called a \emph{proposal} and an \emph{involution} (a function that is its own inverse) that maps tuples of the form $(t, u)$ to tuples of the form $(t', u')$ where $t$ and $t'$ are traces of the model generative function, and $u$ and $u'$ are traces of the proposal generative function.
+In general, both $t$ and $u$ (and by necessity $t'$ and $u'$) may contain the values of continuous random choices, in which case the involution must be constructed from a family of differentiable bijections on the values of these continuous random choices.
+When applying a particular RJMCMC kernel, the determinant of the Jacobian of one of these bijections is required for the acceptance probability calculation.
+This document describes how to implement this Jacobian determinant computation efficiently using automatic differentiation, while exploiting the sparsity of the Jacobian.
+The approach uses a domain-specific language (DSL) for specifying families of bijections that makes the sparsity structure of the Jacobian explicit.
+
+\section{Example}
+
+We consider a variant of the changepoint model that was used to introduce RJMCMC in~\citep{green1995reversible}.
+Briefly, we put a prior on a class of piecewise constant functions on an interval $[0, x_{max}]$ where the number of segments is random and Poisson distributed.
```

Comment:
num segments is Poisson + 1, num changepoints is Poisson?

Reply:
Done